ext/semian/sysv_semaphores.c

```diff
@@ -218,28 +204,22 @@ initialize_new_semaphore_values(int sem_id, long permissions)
   if (semctl(sem_id, 0, SETALL, init_vals) == -1) {
     raise_semian_syscall_error("semctl()", errno);
   }
-#ifdef DEBUG
-  print_sem_vals(sem_id);
-#endif
+  dprint_sem_vals("initialize_new_semaphore_values", sem_id);
 }
 
 static int
-wait_for_new_semaphore_set(key_t key, long permissions)
+wait_for_new_semaphore_set(uint64_t key, long permissions)
 {
-  int i;
-  int sem_id = -1;
-  union semun sem_opts;
   struct semid_ds sem_ds;
-
+  union semun sem_opts;
   sem_opts.buf = &sem_ds;
-  sem_id = semget(key, 1, permissions);
 
+  int sem_id = semget(key, 1, permissions);
```

Comment:
Does this produce a warning? I thought that `key_t` was a 32-bit value and you've changed the type of 64 bits.

Reply:
I didn't see one, and I compile with `-Wall`.
All the values are generated with:

```c
key_t generate_key(const char *name);
```

I was following what was done in `semian_resource_t`.